Source/WebCore/page/cocoa/ResourceUsageThreadCocoa.mm

```diff
@@ -160,8 +160,9 @@ static unsigned categoryForVMTag(unsigned tag)
         return;
     }
 
-    // Main thread is always first.
-    ASSERT(threads[0].dispatchQueueName == "com.apple.main-thread"_s);
+    // Main thread should be always first. If not, just give up for collection.
+    if (threads[0].dispatchQueueName != "com.apple.main-thread"_s)
+        return;
 
```

Comment:
I don't understand this change. We know that this must not happen, but it does. So why remove the assertion, and change to silently fail? This doesn't fix anything, but makes diagnosing the problem harder.

Reply:
I know this is just a workaround, but actually this does happen and it prevents my daily development flow. I don't know anything about mach yet so I cannot fix this. I'm happy to delegate this to experts but I want the fix ASAP.

Reply:
At lease, if assertion is wrong, change it to early return is the best thing we can do without using a brain. It crashes with current code. Then open a new bug that to fix the bug threadInfo() sometimes not returning main thread. How about this?

Reply:
If you are running into this daily, seems like this bug should just be fixed. The point of the assertion is that we couldn't reproduce, and needed more information about the cause. How do you reproduce?